modules/vaos/spec/services/appointment_service_spec.rb

```diff
@@ -5,12 +5,55 @@
 describe VAOS::AppointmentService do
   subject { described_class.for_user(user) }
 
-  let(:user) { build(:user, :mhv) }
+  let(:user) { build(:user, :vaos) }
   let(:start_date) { Time.zone.parse('2019-11-14T07:00:00Z') }
   let(:end_date) { Time.zone.parse('2020-03-14T08:00:00Z') }
 
   before { allow_any_instance_of(VAOS::UserService).to receive(:session).and_return('stubbed_token') }
 
+  describe '#put_cancel_appointment' do
+    context 'when appointment cannot be cancelled' do
+      let(:request_body) do
+        {
+          appointment_time: '11/15/19 20:00:00',
+          clinic_id: '408',
+          cancel_reason: 'whatever',
+          cancel_code: '5',
+          remarks: nil,
+          clinic_name: nil
+        }
+      end
+
+      it 'returns the bad request with detail in errors' do
+        VCR.use_cassette('vaos/appointments/put_cancel_appointment_400', match_requests_on: %i[method uri]) do
+          expect { subject.put_cancel_appointment(request_body) }.to raise_error(
+            Common::Exceptions::BackendServiceException
+          )
+        end
+      end
+    end
+
+    context 'when appointment can be cancelled' do
```

Comment:
Still not clear to me what makes the difference here between can and can't cancel.

Reply:
We are writing a request spec that tests that the exception that is raised from the service, properly renders the error that is under test here. In particular, that detail matches what the service returned in the response body, that our middlewares effectively handled things.